security/advancedtls/advancedtls.go

```diff
@@ -90,14 +102,24 @@ type ClientOptions struct {
 	// function every time asked to present certificates to the server when a new connection is
 	// established. This is known as peer certificate reloading.
 	GetClientCertificate func(*tls.CertificateRequestInfo) (*tls.Certificate, error)
-	// VerifyPeer is a custom server authorization checking after certificate signature check.
-	// If this is set, we will replace the hostname check with this customized authorization check.
-	// If this is nil, we fall back to typical hostname check.
+	// VerifyPeer is a custom verification check after certificate signature check.
+	// If this is set, we will perform this customized check after doing the normal check(s)
+	// indicated by setting |VType|.
 	VerifyPeer CustomVerificationFunc
 	// ServerNameOverride is for testing only. If set to a non-empty string,
 	// it will override the virtual host name of authority (e.g. :authority header field) in requests.
 	ServerNameOverride string
+	// RootCertificateOptions is REQUIRED to be correctly set on client side.
 	RootCertificateOptions
+	// If setting this field to |CertAndHostVerification|, we would perform both the certificate
+	// verification and hostname verification.
+	// If setting to |CertVerification|, we would only perform verification for the certificate sent from
+	// the server. Setting this field without proper custom verification check would leave the
+	// application to the MITM attack.
```

Comment:
leave the application "susceptible" to the MITM attack?

Reply:
Done.